substrate/frame/staking/src/benchmarking.rs

```diff
@@ -769,94 +879,135 @@ benchmarks! {
-			era: EraIndex::zero(),
-		};
-		for _ in 0 .. l {
+		let unlock_chunk =
+			UnlockChunk::<BalanceOf<T>> { value: 1u32.into(), era: EraIndex::zero() };
+		for _ in 0..l {
 			staking_ledger.unlocking.try_push(unlock_chunk.clone()).unwrap();
 		}
 		Ledger::<T>::insert(controller, staking_ledger);
 		let slash_amount = asset::existential_deposit::<T>() * 10u32.into();
 		let balance_before = asset::stakeable_balance::<T>(&stash);
-	}: {
-		crate::slashing::do_slash::<T>(
-			&stash,
-			slash_amount,
-			&mut BalanceOf::<T>::zero(),
-			&mut NegativeImbalanceOf::<T>::zero(),
-			EraIndex::zero()
-		);
-	} verify {
+
+		#[block]
+		{
+			crate::slashing::do_slash::<T>(
+				&stash,
+				slash_amount,
+				&mut BalanceOf::<T>::zero(),
+				&mut NegativeImbalanceOf::<T>::zero(),
+				EraIndex::zero(),
+			);
+		}
+
 		let balance_after = asset::stakeable_balance::<T>(&stash);
 		assert!(balance_before > balance_after);
+
+		Ok(())
 	}
 
-	get_npos_voters {
+	#[benchmark]
+	fn get_npos_voters(
 		// number of validator intention. we will iterate all of them.
-		let v in (MaxValidators::<T>::get() / 2) .. MaxValidators::<T>::get();
-		// number of nominator intention. we will iterate all of them.
-		let n in (MaxNominators::<T>::get() / 2) .. MaxNominators::<T>::get();
+		v: Linear<{ MaxValidators::<T>::get() / 2 }, { MaxValidators::<T>::get() }>,
 
-		let validators = create_validators_with_nominators_for_era::<T>(
-			v, n, MaxNominationsOf::<T>::get() as usize, false, None
-		)?
-		.into_iter()
-		.map(|v| T::Lookup::lookup(v).unwrap())
-		.collect::<Vec<_>>();
+		// number of nominator intention. we will iterate all of them.
+		n: Linear<{ MaxNominators::<T>::get() / 2 }, { MaxNominators::<T>::get() }>,
+	) -> Result<(), BenchmarkError> {
+		create_validators_with_nominators_for_era::<T>(
+			v,
+			n,
+			MaxNominationsOf::<T>::get() as usize,
+			false,
+			None,
+		)?;
 
 		assert_eq!(Validators::<T>::count(), v);
 		assert_eq!(Nominators::<T>::count(), n);
 
 		let num_voters = (v + n) as usize;
-	}: {
+
 		// default bounds are unbounded.
-		let voters = <Staking<T>>::get_npos_voters(DataProviderBounds::default());
+		let voters;
+		#[block]
+		{
+			voters = <Staking<T>>::get_npos_voters(DataProviderBounds::default());
+		}
+
 		assert_eq!(voters.len(), num_voters);
+
+		Ok(())
 	}
 
-	get_npos_targets {
+	#[benchmark]
+	fn get_npos_targets(
 		// number of validator intention.
-		let v in (MaxValidators::<T>::get() / 2) .. MaxValidators::<T>::get();
+		v: Linear<{ MaxValidators::<T>::get() / 2 }, { MaxValidators::<T>::get() }>,
+	) -> Result<(), BenchmarkError> {
 		// number of nominator intention.
 		let n = MaxNominators::<T>::get();
 
-		let _ = create_validators_with_nominators_for_era::<T>(
-			v, n, MaxNominationsOf::<T>::get() as usize, false, None
-		)?;
-	}: {
+		#[block]
+		{
+			create_validators_with_nominators_for_era::<T>(
+				v,
+				n,
+				MaxNominationsOf::<T>::get() as usize,
+				false,
+				None,
+			)?;
```

Comment:
This is wrong, we should benchmark for the lines below.

Reply:
Thanks for catching this, and opening a PR.